src/utils/waitUtils.ts

```diff
@@ -0,0 +1,34 @@
+export const waitAndRetry = async <T>(
+  predicateFn: () => T,
+  sleepTime = 20,
+  maxRetryCount = 15,
+): Promise<T> => {
+  return new Promise((resolve, reject) => {
+    let retryCount = 0
+    function performCheck() {
+      // amount of retries exceeded
+      if (maxRetryCount !== 0 && retryCount > maxRetryCount) {
```

Comment:
🟢 I know I'm late, but this definitely warrants a JSDoc comment to explain that passing `maxRetryCount` as `0` is equivalent to having no maximum.

Reply:
Although I'm not convinced that `0` meaning no limit is useful when JS has a perfectly good `Infinity` numeric.

Reply:
will add jsdoc

I've never seen any library using Infinity, tbh